source/extensions/filters/network/echo/echo.cc

```diff
@@ -0,0 +1,61 @@
+#include "extensions/filters/network/echo/echo.h"
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/network/connection.h"
+#include "envoy/registry/registry.h"
+#include "envoy/server/filter_config.h"
+
+#include "common/common/assert.h"
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilter {
+namespace Echo {
+
+Network::FilterStatus Instance::onData(Buffer::Instance& data, bool end_stream) {
+  ENVOY_CONN_LOG(trace, "echo: got {} bytes", read_callbacks_->connection(), data.length());
+  read_callbacks_->connection().write(data, end_stream);
+  ASSERT(0 == data.length());
+  return Network::FilterStatus::StopIteration;
+}
+
+/**
+ * Config registration for the echo filter. @see NamedNetworkFilterConfigFactory.
+ */
+class EchoConfigFactory : public Server::Configuration::NamedNetworkFilterConfigFactory {
```

Comment:
Originally I liked the plan of moving the factory in with the .cc code but if we're creating a directory per filter I'd lean towards keeping them split out as for filters with a bunch of functionality the filler template code can then be more easily ignored.  Mild preference and I don't remember who asked for the combination to add them in to the conversation heer.

Reply:
I just moved it into the same file, but agreed it seems fine to have it be a different file. I will update and split the config out to a dedicated cc file (but keep it in the same directory).